test/testutils/test_helper.cpp

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright (C) by Erik Verbruggen <erik@verbruggen.consulting>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include <QCoreApplication>
+#include <QDir>
+#include <QFile>
+
+#include <iostream>
+#include <vector>
+
+#include <libsync/filesystem.h>
+
+using namespace std;
+
+/**
+ * @brief The abstract Command class. You know, from the pattern.
+ */
+class Command
+{
+public:
+    Command(const QString &fileName)
+        : _fileName(fileName)
+    {
+    }
+    virtual ~Command() = 0;
+
+    virtual bool execute(QDir &rootDir) const = 0;
+
+protected:
+    static QString parseFileName(QStringListIterator &it)
+    {
+        return it.next();
+    }
+
+    const QString _fileName;
+};
+
+Command::~Command() { }
+
+class SetMtimeCommand : public Command
+{
+public:
+    static const QString name;
```

Comment:
Hmm I guess we could use `constexpr std::string_view name` then the name could be inline?
(I don't like the name to be defined outside of the class).

Reply:
done